lib/handlebars/runtime.js

```diff
@@ -69,12 +69,28 @@ export function template(templateSpec, env) {
       if (!(name in obj)) {
         throw new Exception('"' + name + '" not defined in ' + obj);
       }
-      return obj[name];
+      return container.lookupProperty(obj, name);
```

Comment:
It appears `strict` mode is already covered by the check in `javascript-compiler.js`:

https://github.com/handlebars-lang/handlebars.js/blob/16bd606fec0ada4c811eb7afe6bfd619d88f3d37/lib/handlebars/compiler/javascript-compiler.js#L16-L19

However, the reason why this vulnerability appeared in 4.x was because that check was removed and this change was present. Better to make this change to make it completely clear.